packages/beacon-node/src/api/impl/events/index.ts

```diff
@@ -33,35 +35,42 @@ export function getEventsApi({chain, config}: Pick<ApiModules, "chain" | "config
     ) => routes.events.EventData[K][];
   } = {
     [routes.events.EventType.head]: (data) => [data],
-    [routes.events.EventType.block]: (block) => [
-      {
-        block: toHexString(config.getForkTypes(block.message.slot).BeaconBlock.hashTreeRoot(block.message)),
-        slot: block.message.slot,
-        executionOptimistic: IS_OPTIMISTIC_TEMP,
-      },
-    ],
+    [routes.events.EventType.block]: (block) => {
+      const blockRoot = config.getForkTypes(block.message.slot).BeaconBlock.hashTreeRoot(block.message);
+      const blockSummary = forkchoice.getBlock(blockRoot);
+      return [
+        {
+          block: toHexString(blockRoot),
+          slot: block.message.slot,
+          executionOptimistic: blockSummary != null && isOptimisticBlock(blockSummary),
+        },
+      ];
+    },
     [routes.events.EventType.attestation]: (attestation) => [attestation],
     [routes.events.EventType.voluntaryExit]: (block) => Array.from(block.message.body.voluntaryExits),
     [routes.events.EventType.finalizedCheckpoint]: (checkpoint, state) => [
       {
         block: toHexString(checkpoint.root),
         epoch: checkpoint.epoch,
         state: toHexString(state.hashTreeRoot()),
-        executionOptimistic: IS_OPTIMISTIC_TEMP,
-      },
-    ],
-    [routes.events.EventType.chainReorg]: (oldHead, newHead, depth) => [
-      {
-        depth,
-        epoch: computeEpochAtSlot(newHead.slot),
-        slot: newHead.slot,
-        newHeadBlock: newHead.blockRoot,
-        oldHeadBlock: oldHead.blockRoot,
-        newHeadState: newHead.stateRoot,
-        oldHeadState: oldHead.stateRoot,
-        executionOptimistic: IS_OPTIMISTIC_TEMP,
+        executionOptimistic: false,
       },
     ],
+    [routes.events.EventType.chainReorg]: (oldHead, newHead, depth) => {
+      const blockSummary = forkchoice.getBlock(fromHexString(newHead.blockRoot));
```

Comment:
@tuyennhv if you change the type of Chain.reorg to match EventType.chainReorg, you won't need to do this conversion, and won't have to do forkchoice calls in this event transformation.